Terminal.Gui/View/ViewKeyboard.cs

```diff
@@ -1,464 +1,869 @@
+			if (AddKeyBindingsForHotKey (_hotKey, value)) {
+				// This will cause TextFormatter_HotKeyChanged to be called, firing HotKeyChanged
+				_hotKey = TextFormatter.HotKey = value;
 			}
 		}
+	}
 
-		/// <summary>
-		/// Gets or sets the specifier character for the hotkey (e.g. '_'). Set to '\xffff' to disable hotkey support for this View instance. The default is '\xffff'. 
-		/// </summary>
-		public virtual Rune HotKeySpecifier {
-			get {
-				if (TextFormatter != null) {
-					return TextFormatter.HotKeySpecifier;
-				} else {
-					return new Rune ('\xFFFF');
-				}
+	/// <summary>
+	/// Adds key bindings for the specified HotKey. Useful for views that contain multiple items that each have their own HotKey
+	/// such as <see cref="RadioGroup"/>.
+	/// </summary>
+	/// <remarks>
+	/// <para>
+	/// By default key bindings are added for both the base key (e.g. <see cref="Key.D3"/>) and
+	/// the Alt-shifted key (e.g. <see cref="Key.D3"/> | <see cref="Key.AltMask"/>).
+	/// This behavior can be overriden by overriding <see cref="AddKeyBindingsForHotKey"/>.
+	/// </para>
+	/// <para>
+	/// By default, when <paramref name="hotKey"/> is <see cref="Key.A"/> through <see cref="Key.Z"/> key bindings will be added for both the un-shifted and shifted
+	/// versions. This means if the HotKey is <see cref="Key.A"/>, key bindings for <see cref="Key.A"/> and <see cref="Key.A"/> | <see cref="Key.ShiftMask"/>
+	/// will be added. This behavior can be overriden by overriding <see cref="AddKeyBindingsForHotKey"/>.
+	/// </para>
+	/// <para>
+	/// For each of the bound keys <see cref="Command.Default"/> causes the view to be focused and <see cref="Command.Accept"/> does nothing.
+	/// </para>
+	/// </remarks>
+	/// <param name="prevHotKey">The HotKey <paramref name="hotKey"/> is replacing. Key bindings for this key will be removed.</param>
+	/// <param name="hotKey">The new HotKey. If <see cre="Key.Null"/> <paramref name="prevHotKey"/> bindings will be removed.</param>
+	/// <returns><see langword="true"/> if the HotKey bindings were added.</returns>
+	/// <exception cref="ArgumentException"></exception>
+	public virtual bool AddKeyBindingsForHotKey (Key prevHotKey, Key hotKey)
+	{
+		if (_hotKey == hotKey) {
+			return false;
+		}
+
+		var newKey = hotKey == Key.Unknown ? Key.Null : hotKey;
+
+		var baseKey = newKey & ~Key.CtrlMask & ~Key.AltMask & ~Key.ShiftMask;
+		if (newKey != Key.Null && (baseKey == Key.Space || Rune.IsControl (KeyEventArgs.ToRune (baseKey)))) {
+			throw new ArgumentException (@$"HotKey must be a printable (and non-space) key ({hotKey}).");
+		}
+
+		if (newKey != baseKey) {
+			if ((newKey & Key.CtrlMask) != 0) {
+				throw new ArgumentException (@$"HotKey does not support CtrlMask ({hotKey}).");
 			}
-			set {
-				TextFormatter.HotKeySpecifier = value;
-				SetHotKey ();
+			// Strip off the shift mask if it's A...Z
+			if (baseKey is >= Key.A and <= Key.Z && (newKey & Key.ShiftMask) != 0) {
+				newKey &= ~Key.ShiftMask;
```

Comment:
Just a thought here but is `char.IsUpper` a better fit here? would it make this functionality more language independent? e.g. accented characters etc.

Reply:
Nope, I don't think so. Remember `Key.A` is 65, which is upper case. But `(Key)'a'` is 97 and lower case. As is `(Key)â`. If we used `char.IsUpper` in cases like this `(Key)â` would become `(Key)Â` which is not what we want.